src/bfx_tools/vaxrank.ml

```diff
@@ -0,0 +1,150 @@
+               [--max-mutations-in-report MAX_MUTATIONS_IN_REPORT]
+*)
+module Configuration = struct
+  type t = {
+    name: string;
+    (* vaxrank-specific ones *)
+    vaccine_peptide_length: int;
+    padding_around_mutation: int;
+    max_vaccine_peptides_per_mutation: int;
+    max_mutations_in_report: int;
+    (* isovar-like ones *)
+    min_mapping_quality: int;
+    min_reads_supporting_variant_sequence: int;
+    use_duplicate_reads: bool;
+    drop_secondary_alignments: bool;
+    (* topiary-like ones *)
+    mhc_epitope_lengths: int list;
+    (* the rest *) 
+    parameters: (string * string) list;
+  }
+  let to_json {
+    name; 
+    vaccine_peptide_length; 
+    padding_around_mutation; 
+    max_vaccine_peptides_per_mutation; 
+    max_mutations_in_report; 
+    min_mapping_quality;
+    min_reads_supporting_variant_sequence;
+    use_duplicate_reads;
+    drop_secondary_alignments;
+    mhc_epitope_lengths;
+    parameters}: Yojson.Basic.json 
+    =
+    `Assoc [
+      "name", `String name;
+      "vaccine_peptide_length", `Int vaccine_peptide_length;
+      "padding_around_mutation", `Int padding_around_mutation;
+      "max_vaccine_peptides_per_mutation", `Int max_vaccine_peptides_per_mutation;
+      "max_mutations_in_report", `Int max_mutations_in_report;
+      "min_mapping_quality", `Int min_mapping_quality;
+      "min_reads_supporting_variant_sequence", 
+        `Int min_reads_supporting_variant_sequence;
+      "use_duplicate_reads", `Bool use_duplicate_reads;
+      "drop_secondary_alignments", `Bool drop_secondary_alignments;
+      "mhc_epitope_lengths", 
+        `List (List.map mhc_epitope_lengths ~f:(fun i -> `Int i));
+      "parameters",
+      `Assoc (List.map parameters ~f:(fun (a, b) -> a, `String b));
+    ]
+  let render {parameters; _} =
+    List.concat_map parameters ~f:(fun (a,b) -> [a; b])
+
+  let default =
+    {name = "default";
+     vaccine_peptide_length = 25; 
+     padding_around_mutation = 0; 
+     max_vaccine_peptides_per_mutation = 1;
+     max_mutations_in_report = 10;
+     min_mapping_quality = 1;
+     min_reads_supporting_variant_sequence = 2;
+     use_duplicate_reads = false;
+     drop_secondary_alignments = false;
+     mhc_epitope_lengths = [8; 9; 10; 11];
+     parameters = []}
+  let name t = t.name
+end
+
+let run ~(run_with: Machine.t)
+  ~configuration 
+  ~reference_build
+  ~vcf
+  ~bam
+  ~predictor 
+  ~alleles_file
+  ~output_folder
+  =
+  let open KEDSL in
+  let open Configuration in
+  let c = configuration in
+  let vaxrank =
+    Machine.get_tool run_with Machine.Tool.Definition.(create "vaxrank")
+  in
+  let var_arg = ["--vcf"; vcf#product#path] in
+  let predictor_arg = 
+    ["--mhc-predictor"; (Topiary.predictor_to_string predictor)] in
+  let allele_arg = ["--mhc-alleles-file"; alleles_file#product#path] in
+  let soi = string_of_int in
+  let vax_length_arg = ["--vaccine-peptide-length"; soi c.vaccine_peptide_length] in
```

Comment:
All of these c.xxx options that are coming from the configuration should be in `Configuration.render` (I know, it's super tedious to enumerate them so many times in the config, one of the least-fun parts of adding a tool)